src/components/button/button.component.tsx

```diff
@@ -115,12 +120,21 @@ function renderChildren({
   tooltipTarget,
 }: /* eslint-enable */
 RenderChildrenProps) {
-  const iconColorMap = {
-    primary: "--colorsActionMajorYang100",
-    secondary: "--colorsActionMajor500",
-    tertiary: "--colorsActionMajor500",
-    darkBackground: "--colorsActionMajor500",
-    dashed: "--colorsActionMajor500",
+  const iconColor = () => {
```

Comment:
**comment** (non-blocking): I actually preferred the previous approach of using an object/map to a `switch` statement. But it's obviously something you cared about (since there was no need to change it) and I don't care enough to block the PR over it.

Reply:
Tbh it was just to avoid having to declare the same colour for both gradient variants, I'm happy to revert it back if the second reviewer also prefers the old approach

Reply:
oh I hadn't even noticed the repetition 🤦 That's definitely a small argument in favour of your change - overall I still slightly prefer the object but as I said I can easily live with it either way, happy to leave it up to you and/or a second reviewer.

Reply:
No worries Tom has pointed out that there's other repetition there as well 🤦, I'll update it based on his suggestion but likely go with `if` statements as it can be shorthanded a bit more that way

```
const iconColor = () => {
    if (buttonType === "primary") {
      return "--colorsActionMajorYang100";
    }

    if (buttonType.includes("gradient")) {
      return "--colorsActionMinorYin090";
    }

    return "--colorsActionMajor500";
  };